packages/eas-json/src/utils.ts

```diff
@@ -23,6 +23,26 @@ export class EasJsonUtils {
     return resolveBuildProfile({ easJson, platform, profileName });
   }
 
+  public static getBuildProfileDepreactionWarnings(
+    buildProfile: BuildProfile,
+    profileName?: string
+  ): { message: string[]; docsUrl?: string }[] {
+    const warnings: { message: string[]; docsUrl?: string }[] = [];
```

Comment:
IMO it deserves a separate type
```suggestion
  ): Xyz[] {
    const warnings: Xyz[] = [];
```